jvm/hermes/src/main/jni/RuntimeScope.cpp

```diff
@@ -0,0 +1,95 @@
+#include "RuntimeScope.h"
+#include <jsi/jsi.h>
+#include <stdexcept>
+#include "iostream"
+
+namespace intuit::playerui {
+
+//template<typename T>
+void RuntimeScope::trackValue(void* ptr, Value value) {
+    valueScope->insert({ptr, make_unique<Value>(std::move(value))});
+}
+
+void RuntimeScope::trackFunction(void* ptr, Function value) {
+    functionScope->insert({ptr, make_unique<Function>(std::move(value))});
+}
+
+void RuntimeScope::trackArray(void *ptr, facebook::jsi::Array value) {
+    arrayScope->insert({ptr, make_unique<Array>(std::move(value))});
+}
+
+void RuntimeScope::trackObject(void *ptr, facebook::jsi::Object value) {
+    objectScope->insert({ptr, make_unique<Object>(std::move(value))});
+}
+
+void RuntimeScope::trackSymbol(void *ptr, facebook::jsi::Symbol value) {
+    symbolScope->insert({ptr, make_unique<Symbol>(std::move(value))});
+}
+
+Value* RuntimeScope::getValue(void* ptr) {
+    auto it = valueScope->find(ptr);
+    if (it != valueScope->end()) {
+        return valueScope->at(ptr).get();
+    }
+    throw std::runtime_error("Value was not found in the RuntimeScope");
```

Comment:
Why throw here? The return type is nullable, it should just return `nullptr`.

Otherwise, change return type to `Value&` to be more accurate (although, my preference would be keep nullable return type to match standard map APIs and let the caller handle `nullptr` explicitly.

Reply:
this would apply to all the pointers, right?